staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1363,3 +1369,27 @@ func (e *Store) ConvertToTable(ctx context.Context, object runtime.Object, table
 func (e *Store) StorageVersion() runtime.GroupVersioner {
 	return e.StorageVersioner
 }
+
+// validatePredicateFunc will validate the collision of indexes(including the label index and field index).
+func validatePredicateFunc(fn func(label labels.Selector, field fields.Selector) storage.SelectionPredicate) error {
```

Comment:
Thanks, this is an improvement, and you should keep this function in any case to prevent double-registration.

However I think it would be even better if label names and field names were prefixed (e.g., with "l:" and "f:" respectively) to produce the index name. That would make it very obvious that no collision is possible.

Reply:
> Thanks, this is an improvement, and you should keep this function in any case to prevent double-registration.

this function will only be called once for every resource(pod/node).

> However I think it would be even better if label names and field names were prefixed (e.g., with "l:" and "f:" respectively) to produce the index name. That would make it very obvious that no collision is possible.

@wojtek-t what do you think?

Reply:
This is pretty much exactly what I proposed above:
https://github.com/kubernetes/kubernetes/pull/85445#discussion_r362003025

So yes - I completely agree with Daniel.

Reply:
@shaloulcy - will you have time to fix that some time in the next couple days?

Reply:
@wojtek-t @lavalamp PTAL